homeassistant/components/simplisafe/__init__.py

```diff
@@ -511,9 +513,15 @@ async def refresh(event_time):
             """Refresh data from the SimpliSafe account."""
             await self.async_update()
 
+        interval = timedelta(
+            seconds=self._config_entry.options.get(
+                CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL
+            )
+        )
+
         self._hass.data[DOMAIN][DATA_LISTENER][
             self._config_entry.entry_id
-        ] = async_track_time_interval(self._hass, refresh, DEFAULT_SCAN_INTERVAL)
+        ] = async_track_time_interval(self._hass, refresh, interval)
```

Comment:
We don't allow configuring of scan intervals view options.

Reply:
Why is that?

Reply:
There's nothing preventing people from abusing the SimpliSafe API by setting a low value. This is why a reasonable default of 30 seconds was set.

Reply:
ok, and is there no way to set a minimum?

Reply:
I added a minimum value

Reply:
We have no indication that SimpliSafe will respond favorably if hundreds (or more) of HASS instances start pinging their Cloud more than every 30 seconds. I'm not comfortable with making the scan interval configurable until there's solid data supporting that we can do this safely.

Reply:
How can we get data without having users try it? Also, I built the Homebridge equivalent (https://github.com/nzapponi/homebridge-simplisafe3), the default there is 15 seconds and it works well (I probably have one rate-limiting event per month, if that day I have been using the SimpilSafe mobile app a lot).

There are plenty of plugins where you can set refresh rates, which could cause all sorts of issues with billing and rate limiting (see most weather plugins for instance), I don't see why it's a problem to add it here.

Reply:
Here is an example of what happens if we don't set boundaries on rate limits https://github.com/home-assistant/core/issues/41352.  Another example is when august started imposing very strict rate limits and broke every's lock integration.